packages/config/src/project/index.ts

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash'
+import Debug from 'debug'
+import deepDiff from 'return-deep-diff'
+
+import errors, { ConfigValidationFailureInfo, CypressError } from '@packages/errors'
+import { getCtx } from '@packages/data-context/src/globalContext'
+import type {
+  ResolvedFromConfig, TestingType, FullConfig,
+} from '@packages/types'
+
+import {
+  validate,
+  validateNoBreakingConfig,
+} from '../browser'
+import {
+  setPluginResolvedOn,
+  mergeDefaults,
+} from './utils'
+
+const debug = Debug('cypress:config:project')
+
+// TODO: any -> SetupFullConfigOptions in data-context/src/data/ProjectConfigManager.ts
```

Comment:
Using `data-context/src/data/ProjectConfigManager.ts` type definitions breaks the type. It seems that some types are missing. I postponed it for now because I need to read a lot of code to be sure that the new types are correct.